src/main/java/net/sf/jabref/gui/JabRefFrame.java

```diff
@@ -641,6 +644,7 @@ public void windowClosing(WindowEvent e) {
             bp.updateSearchManager();
             // Set correct enabled state for Back and Forward actions:
             bp.setBackAndForwardEnabledState();
+            bp.getUndoManager().postUndoRedoEvent();
```

Comment:
Oh, it is actually not in `windowClosing()`. It is in the `tabbedPane`  changeListener in `init()`.